<?xml version="1.0"?>

<launch>
  <arg name="world" default="$(find urs_wearable)/worlds/empty.world" />
  <arg name="reference_latitude" default="49.860246" />
  <arg name="reference_longitude" default="8.687077" />
  <arg name="uav_total" default="4" />
  <arg name="uav_ns" default="/uav" />

  <include file="$(find hector_gazebo_worlds)/launch/start.launch">
    <arg name="world" value="$(arg world)" />
    <arg name="gui" value="true" />
  </include>

  <include file="$(find urs_wearable)/launch/spawn_quadrotors.launch">
    <arg name="reference_latitude" value="$(arg reference_latitude)" />
    <arg name="reference_longitude" value="$(arg reference_longitude)" />
  </include>

  <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
    <arg name="port" value="9090"/>
  </include>

  <node pkg="urs_wearable" type="exec_monitor" name="exec_monitor" output="screen">
    <param name="uav_total" value="$(arg uav_total)" type="int" />
	  <param name="uav_ns" value="$(arg uav_ns)" type="str" />

	  <param name="domain_file" value="$(find urs_wearable)/../FF-v2.3/urs_d.pddl" type="str" />
	  <param name="planner_command" value="timeout 10 $(find urs_wearable)/../FF-v2.3/ff" type="str" />
	  <param name="problem_path" value="$(find urs_wearable)/../FF-v2.3/" type="str" />
	  <param name="tmp_path" value="$(find urs_wearable)/../tmp/" type="str"/>
  </node>

  <!-- Start rviz visualization with preset config -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find urs_wearable)/rviz_cfg/four_quadrotors.rviz" />
</launch>
